### ISSUE TITLE
When log file doesn't exist, it should be automatically created

### ISSUE THREAD

**Issue by kadishmal** [nomiddlename/log4js-node#121]
Since at the initial stage of the app, no log files exist, log4js needs to create required log files automatically without failing.

Since no log files existed on my clean machine, the following error was constantly displayed:

```
log4js.fileAppender - Writing to file /XXXXXXX/assets/logs/XXXX.log, error happened  { [Error: ENOENT, open '/XXXXXXX/assets/logs/XXXX.log']
  errno: 34,
  code: 'ENOENT',
  path: '/XXXXXXX/assets/logs/XXXX.log' }
```

**Comment by nomiddlename** [comment]
This issue has been fixed a while ago.

**Comment by RenaldasK** [comment]
Which version is a "while ago"? I'd like to upgrade as I'm having the same issue.

**Comment by nomiddlename** [comment]
Log4js uses streamroller to handle file rolling - making directories has been included in that since v0.4.0 (see [this commit](https://github.com/log4js-node/streamroller/commit/44ee05078a2bf0fa2ddef91b27c095fb50a0a4b1) ). That version or later has been included in all the log4js 2.x releases.
